vllm/v1/core/scheduler.py

```diff
@@ -311,6 +344,10 @@ def schedule(self) -> "SchedulerOutput":
                         self.encoder_cache_manager.allocate(request, i)
                     encoder_budget = new_encoder_budget
 
+        # Put back any skipped requests at the head of the waiting queue
+        if waiting_for_fsm:
+            self.waiting.extendleft(waiting_for_fsm)
```

Comment:
@aarnphm extendleft() is not what you want here. It reverses the order of elements.

https://docs.python.org/3/library/collections.html#collections.deque.extendleft

Reply:
This will work if you change `append()` to `appendleft()` when elements are added. That seems a bit more confusing to me, but it'll be correct.

Reply:
Oh i forgot to reversed the list b4 extendlefy

Reply:
this is fixed -- changed the earlier `append()` to `appendleft()` so priority order is retained among the blocked guided requests